### PR TITLE
Swagger endpoint

### PR DESCRIPTION
**Pull request by Adrien-COSTANDI**

### CODE REVIEW

File: src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java [hunk 1]
```diff
@@ -68,34 +71,34 @@ public List<ConfigurationGeneralDto> getAllConfiguration() {
...
-          required = true)
+      @io.swagger.v3.oas.annotations.parameters.RequestBody(
+            description = "The configuration to send to the controller.",
+            required = true,
+            content = @Content(
+                  examples = @ExampleObject(
+                        """
+                        {
+                          "name": "string",
+                          "description": "string",
```
Comment (tvelu77) [inline src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java]: Maybe, explain what a description is ?

File: src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java [hunk 2]
```diff
@@ -68,34 +71,34 @@ public List<ConfigurationGeneralDto> getAllConfiguration() {
...
+                          "configuration": "{key1: value1, key2: value2}",
+                          "firmware": 0
```
Comment (tvelu77) [inline src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java:96]: 0 ?
I thought that a firmware was a string like "6.1.1-3232"

File: src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java [hunk 3]
```diff
@@ -68,34 +71,34 @@ public List<ConfigurationGeneralDto> getAllConfiguration() {
...
-                  {
-                    "key": "String",
-                    "value": "String"
-                  }
-                ],
-                "firmware": "String"
-              }""",
-          required = true)
+      @io.swagger.v3.oas.annotations.parameters.RequestBody(
+            description = "The configuration to send to the controller.",
```
Comment (tvelu77) [inline src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java]: "to send" -> "to be sent"

File: src/main/java/fr/uge/chargepointconfiguration/user/UserController.java [hunk 4]
```diff
@@ -36,8 +38,18 @@ public UserController(UserService userService) {
...
+              mediaType = "application/json",
+              schema = @Schema(implementation = UserDto.class)
+        )
+  )
   @GetMapping("/{id}")
-  public UserDto getUserById(@PathVariable int id) {
+  public UserDto getUserById(
+        @Parameter(description = "The user you are looking for.")
+        @PathVariable int id) {
     // TODO : exception BAD REQUEST si id est pas un nombre
```
Comment (tvelu77) [inline src/main/java/fr/uge/chargepointconfiguration/user/UserController.java:53]: A TODO in french, what a shame :(

**Review by tvelu77 — CHANGES_REQUESTED** [review]

File: src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java (hunk 2 above)
Comment (Adrien-COSTANDI) [reply src/main/java/fr/uge/chargepointconfiguration/configuration/ConfigurationController.java:96]: it's a foreign key, referencing the id of a firmware

**Review by tvelu77 — APPROVED** [review]
Very nice !

**Pull request merged**